horace_core/sqw/@sqw/tobyfit_DGdisk_resfun_covariance.m

```diff
@@ -126,20 +120,21 @@
 
     % Compute wavevector-energy covariance matrix in different dimensions
     %TODO: Re #1040 this code is not consistent with generic projections
-    if ~isa(wtmp.data.proj,'line_proj')
+    if ~isa(win(iw).data.proj,'line_proj')
         error('HORACE:sqw:not_implemented', ...
             'resolution cannot currently be calculated for any projection except linear projection')
     end
     
     cov_hkle{iw} = transform_matrix (cov_x, dq_mat);
-    cov_proj{iw} = transform_matrix (cov_hkle{iw}, inv(wtmp.data.u_to_rlu));
+    cov_proj{iw} = transform_matrix (cov_hkle{iw}, inv(win(iw).data.u_to_rlu));
     rlu_to_spec = invert_matrix (spec_to_rlu);
     rlu_to_spec4(1:3,1:3,:) = rlu_to_spec(:,:,irun);
     rlu_to_spec4(4,4,:) = 1;
     cov_spec{iw} = transform_matrix (cov_hkle{iw}, rlu_to_spec4);
 end
 
-if ~iscell(win) && isscalar(win)
+if isscalar(win)
```

Comment:
Can `win` no longer be a base `sqw` array?

Reply:
Previously the code handled the case of win being a cell array of scalar sqw objects, for the case of an ancient possible use case, but this was never used, and is unnecessary anyway as it is so easy to turn a cell array of objects of the same class into an object array (modulo the discussion we had about heterogeneous arrays). The case of a cell array of sqw objects as input cannot access tobyfit_DGdisk_resfun_covariance anyway, as it is in the class definition folder. I therefore have removed what is redundant check.